src/target/cortexm.c

```diff
@@ -650,8 +650,15 @@ bool cortexm_probe(adiv5_access_port_s *ap)
 		/* Release reset */
 		platform_nrst_set_val(false);
 		/* Poll for release from reset */
-		while (target_mem_read32(t, CORTEXM_DHCSR) & CORTEXM_DHCSR_S_RESET_ST)
+		platform_timeout_s timeout;
+		platform_timeout_set(&timeout, 1000);
+		while (target_mem_read32(t, CORTEXM_DHCSR) & CORTEXM_DHCSR_S_RESET_ST) {
+			if (platform_timeout_is_expired(&timeout)) {
+				DEBUG_ERROR("Error releasing from reset\n");
+				break;
```

Comment:
Should this be non-fatal or should it cause probing to fail (that is, `return false;`)? What's the reasoning here please? (Please insert a comment explaining why the behaviour is set how it is just like in adiv5.c's `cortexm_prepare()`)

Reply:
That way, it keeps on trying to detect the chip. Detection may be possible still under reset:
'''> ./blackmagic -t
Black Magic Debug App v1.9.0-1153-gdb3692630
 for Black Magic Probe, ST-Link v2 and v3, CMSIS-DAP, J-Link and FTDI (MPSSE)
Using 1d50:6018 7BAE7AB8 Black Magic Debug
 Black Magic Probe v1.9.0-1153-gdb3692630
Running in Test Mode
Target voltage: 3.4V Volt
Speed set to 1.951MHz for SWD
Switching out of dormant state into SWD
Handling swd multi-drop, TARGETID 0x24820041
DP DPIDR 0x0be12477 (v2 MINDP rev0) designer 0x43b partno 0xbe
TARGETID 0x24820041 designer 0x20 partno 0x4820
AP   0: IDR=14770015 CFG=00000000 BASE=e00fe003 CSW=83800040 (AHB5-AP var1 rev1)
ROM: Table BASE=0xe00fe000 SYSMEM=0x00000001, Manufacturer 020 Partno 482
ROM: Table BASE=0xe00ff000 SYSMEM=0x00000001, Manufacturer 43b Partno 4c9
 0 0xe000e000: Debug component - Cortex-M33 (System Control Space) (PIDR = 0x00000004000bbd21 DEVTYPE = 0x00 ARCHID = 0x2a04)
 -> cortexm_probe
CPUID 0x410fd214 (M33 var 0 rev 4)
Error releasing from reset
Calling stm32f1_probe
Calling stm32f4_probe
Calling stm32h5_probe
Calling stm32h7_probe
Calling stm32l0_probe
Calling stm32l4_probe
ID Code: 00004820
 1 0xe0001000: Debug component - Cortex-M33 (Data Watchpoint and Trace) (PIDR = 0x00000004000bbd21 DEVTYPE = 0x00 ARCHID = 0x1a02)
 2 0xe0002000: Debug component - Cortex-M33 (Breakpoint Unit) (PIDR = 0x00000004000bbd21 DEVTYPE = 0x00 ARCHID = 0x1a03)
 3 0xe0000000: Debug component - Cortex-M33 (Instrumentation Trace Macrocell) (PIDR = 0x00000004000bbd21 DEVTYPE = 0x43 ARCHID = 0x1a01)
 4 Entry 0xfff41002 -> Not present
 5 0xe0041000: Debug component - Cortex-M33 (Embedded Trace) (PIDR = 0x00000004001bbd21 DEVTYPE = 0x13 ARCHID = 0x4a13)
 6 0xe0042000: Debug component - Cortex-M33 (Cross Trigger) (PIDR = 0x00000004000bbd21 DEVTYPE = 0x14 ARCHID = 0x1a14)
 7 Entry 0xfff44002 -> Not present
 ROM: Table END
 1 0xe0040000: 0x00000000 <- does not match preamble (0xb105000d)
 0xe0044000: 0x00000000000a0000 Non ARM component ignored
3 Entry 0x1ff02002 -> Not present
ROM: Table END
SWD access resulted in no response
...
SWD access resulted in no response
***  1   STM32U575/585 M33
Error releasing from reset
Can not attach to target 1'''

Reply:
Right, ok, so the being stuck in reset doesn't actually necessarily hurt the probe and enumeration phase, but instead attach/etc. Now we follow. Thank you.